### PR TITLE
Add git branch to version package

### PR DESCRIPTION
**Pull request by iaroslav-ciupin**
Signed-off-by: Iaroslav Ciupin <iaroslav@union.ai>

## _Read then delete this section_

# TL;DR
Add git branch to version package. Sometimes its  useful for app to know what git branch it was built from.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by katrogan** [comment]
LGTM
@eapolinario @hamersaw @EngHabu can one of you approve for open source?

File: version/version.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ var (
 	Version = "unknown"
 	// Build timestamp
 	BuildTime = time.Now().String()
+	// Git branch that was used to build the binary
+	GitBranch = ""
```
Comment (EngHabu) [inline version/version.go:22]: Let's update the LogBuildInformation() method below to include this (if not empty)

Comment (iaroslav-ciupin) [reply version/version.go:23]: done

File: version/version.go [hunk 2]
```diff
@@ -18,12 +19,18 @@ var (
...
 )
 
 // Use this method to log the build information for the current app. The app name should be provided. To inject the build
 // and version information refer to the top-level comment in this file
 func LogBuildInformation(appName string) {
 	logrus.Info("------------------------------------------------------------------------")
-	logrus.Infof("App [%s], Version [%s], BuildSHA [%s], BuildTS [%s]", appName, Version, Build, BuildTime)
+	msg := fmt.Sprintf("App [%s], Version [%s], BuildSHA [%s], BuildTS [%s]", appName, Version, Build, BuildTime)
+	if GitBranch != "" {
```
Comment (EngHabu) [inline version/version.go:31]: nit:
```suggestion
	if len(GitBranch) > 0 {
```

**Review by EngHabu — APPROVED** [review]

File: version/version.go (hunk 2 above)
Comment (iaroslav-ciupin) [reply version/version.go:31]: don't get why?
